### PR TITLE
chore(types): add point to declaration file

### PR DESCRIPTION
**Pull request by inyourtime**
I tried updating this package, [elysia-swagger](https://github.com/elysiajs/elysia-swagger), to use @scalar/types, but there was an error about a missing declaration file. This PR will fix that.

### CODE REVIEW

**Review by marclave — APPROVED** [review]
@inyourtime ooo awesome, we can ignore the markdown failures but i'll wait on @hanspagel to review this tonight (german time) before we merge : )

**Comment by amritk** [comment]
Hey @inyourtime I think you should be able to import the types from `@scalar/types/legacy`

Sorry there's no documentation right now. Would that work? For our internal packages we usually export from a path

**Comment by inyourtime** [comment]
@amritk I use 

```typescript
import type { ReferenceConfiguration } from '@scalar/types/legacy'
```

So, I get this error
`Cannot find module '@scalar/types/legacy' or its corresponding type declarations.`

tsconfig.json
```json
{
    "compilerOptions": {
        "target": "ES2020",
        "lib": ["ESNext", "DOM", "ScriptHost"],
        "module": "ES2022",
        "moduleResolution": "node",
        "declaration": true,
        "noEmit": true,
        "esModuleInterop": true,
        "forceConsistentCasingInFileNames": true,
        "strict": true,
        "skipLibCheck": true
    }
}
```

**Comment by amritk** [comment]
Hey can you link me the repo, let me give it a try

**Comment by inyourtime** [comment]
@amritk 
I am working on [elysia-swagger](https://github.com/elysiajs/elysia-swagger), but you can work on a package in Scalar that imports `@scalar/types/legacy` and changes `moduleResolution` to `Node`.

**Comment by amritk** [comment]
ooo yes sorry I missed that you are totally right. If you don't want to make any changes to the tsconfig then yea this PR is good, we can just export from the base as well for legacy support.

**Review by amritk — APPROVED** [review]

**Review by hanspagel — APPROVED** [review]
Ah, I’ve had this issue with another package, thanks for enlightening me!! Great PR, thanks for taking the/your time @inyourtime!

**Pull request merged**
